### ISSUE TITLE
Dynamische Rekrutiergeschwindigkeit und Verbesserung der Religion

### ISSUE THREAD

**Issue by Cenraves** [Sh0wdown/UnofficialCrusaderPatch#251]
Ich würde es sehr begrüßen, wenn die Rekrutiergeschwindigkeit je nach dem Gold der KI variiert. Es macht durchaus Sinn, wenn eine KI für Notfälle Gold hortet, damit sie sich zügig wieder aufbauen kann, doch wenn eine gewisse Grenze, zum Beispiel 10000 Gold überschritten werden, sollte das Rekrutiertempo auf das Maximum angehoben werden, da es ab einer bestimmten Menge Gold absurd ist, es nicht in Truppen zu investieren.
AI's wie Richard, die ohnehin die maximale Rekrutiergeschwindigkeit haben, wären davon nicht betroffen. Ich denke, wenn man ihre "normale" Geschwindigkeit runtersetzen würde, hätte das Auswirkungen auf die charakteristischen frühen Angriffe. 
Alternativ, könnte man in Erwägung ziehen, diese Grenzen für jede infrage kommende KI einzeln festzulegen: Mal angenommen die Ratte wäre eine Schlaftablette, würde es bei ihrer Wirtschaftlichkeit weniger Sinn machen, die selbe Grenze wie Friedrich zu erhalten, bis sie in den Turbomodus einschaltet.

Nun zu meinem zweiten Anliegen:
Könnte man auch Anpassungen an der Beliebtheit vornehmen? Eventuell könnte man dann ja den Religionsbonus in sofern verbessern, dass man für +8 Beliebtheit nur noch 15% pro 2 Beliebtheitspunkte der gesegneten Bevölkerung benötigt und man so den Rest nach oben als Puffer offen hat.
Ab einer bestimmten Bevölkerungsdichte, ist es absurd auf den +8er Religionsbonus zu gehen. (permanenter Kathedralen- und Kirchenbonus mal ausgenommen) Gerade im direkten Vergleich mit Bierausschank, steht Religion nicht besonders gut da. Eine Hopfenfarm direkt am Vorratslager reicht aus, um 5-6 Brauereien und 6 Schenken und damit 180 Bevölkerung, für den maximalen Beliebtheitsbonus zu beliefern. Die Kosten dafür würden lediglich ca 215 Holz und 600 Gold betragen. Zurück zum Thema: Wie viele Kapellen müsste man bauen, die jeweils 250 Gold kosten und zusätzlich dazu noch sehr viel Platz einnehmen, um für 180 Leute einen permanenten +8er Bonus zu generieren? Ich denke, das wäre ein sinnloses Unterfangen, da schon ab einer Bevölkerungsdichte von 90 ca 10 Kapellen gebaut werden müssten. Die genauen Zahlen habe ich aber nicht im Kopf, da ich den Religionsbonus so gut wie nie benutzt habe, und das finde ich eben sehr schade.

Getrennte Vorratslager / Kornspeicher / Waffenkammern wären übrigens auch eine Überlegung wert!

Jetzt noch mal ein paar Worte abseits davon:
Ich freue mich sehr darüber zu sehen, wie jemand diesen Rohdiamanten nimmt und ihn zu einem wahren Schmuckstück verarbeitet. Es tat sehr viele Jahre weh, dieses verschwendete Potenzial zu sehen und nichts dagegen tun zu können. Viele der Neuerungen sind für mich in einer Hinsicht schon fast notwendig, aber auch viele Spielereien, die ich mir gewünscht habe, sind endlich dank dir realisiert worden. Gerade der Farbwechsel und die Sache mit den Belagerungsgeräten, waren für mich ein wahrer Schatz. Die heilenden Apotheker waren so eine Sache, die sich sehr viele Leute damals unter Udwins ersten Video dazu gewünscht haben. Dazu dachte ich mir damals noch "Wäre cool, aber wahrscheinlich nicht machbar". Da habe ich mich wohl zum Glück geirrt. Ich finde, alleine schon für diese Leistung, kannst du dir echt selber auf die Schulter klopfen! Vielen Dank, für die Mühe und das Herzblut, was du in dieses Projekt reinsteckst.

**Comment by Lolasik011** [comment]
Thanks @Sh0wdown for  the great patch.

**Comment by Halbarad1** [comment]
Das mit der Truppenproduktion finde ich gut, Religion hingegen ist eben nicht für Monsterwirtschaften geeignet. Wenn du hingegen mal auf einer Wüste ohne Rohstoffe oder Fruchtbarkeiten spielst wird sie auf einmal das non-plus-ultra da nach Erbauung keine weiteren Rohstoffe mehr benötigt werden und nicht einmal das Vorratslager für die Arbeit blockiert werden muss.

**Comment by Sh0wdown** [comment]
Ersteres passiert schon und kann über AIC edits geändert werden: [Wiki](https://github.com/Sh0wdown/UnofficialCrusaderPatch/wiki/AI-Personality)

| | | |
| :---: | :---: | :---: |
| RecruitInterval | Int32 |  |
| RecruitIntervalWeak | Int32 | The 'weak' state sets in if the AI is completely trashed. F.e. troops < 8, gold < 200, population < 15, ... |
| RecruitIntervalStrong | Int32 | The 'strong' state sets in if f.e. the AI has troops >= 40, gold >= 200, population >= 40, ... |

**Comment by Cenraves** [comment]
Vielen lieben Dank dafür! Ich wusste gar nicht, das es so etwas bereits im Spiel gibt. Ich habe bereits etwas rumprobiert, allerdings erreicht man diesen Stärkestatus leider viel zu einfach. Kann man Diesen denn irgendwie neu definieren?

**Comment by Sh0wdown** [comment]
> Vielen lieben Dank dafür! Ich wusste gar nicht, das es so etwas bereits im Spiel gibt. Ich habe bereits etwas rumprobiert, allerdings erreicht man diesen Stärkestatus leider viel zu einfach. Kann man Diesen denn irgendwie neu definieren?

Ja, allerdings nicht über AIC edits. Da müsste ich also eine neue Option für hinzufügen.

**Comment by DarkThief-Darek** [comment]
Ich habe mich ja mittlerweile etwas mit den Vanilla Daten auseinandergesetzt und finde, dass in vielen Fällen der **RecruitIntervalStrong** Parameter eher dazu dient, um die KIs abzuschwächen.

So haben beispielsweise KIs wie der Wolf auf **Strong** mit 4 eine langsamere Rekrutiergeschwindigkeit als auf **Weak** mit 1 (so schnell wie Richard oder Ratte).

Interessanter Ausreißer ist Nizar, der auf **Weak** einen stolzen Wert von 8 hat. Wenn der am Boden ist, bildet er so gut wie keine Truppen aus. ^^

Übrigens werden nur die Werte 0, 1, 4 und 8 vergeben. Ich hoffe, man kann auch alle Werte dazwischen geben.

Frage an @Sh0wdown: ist 0 schneller als 1? Emir beispielsweise hat auf **Default** und **Strong** einen Wert von 0. Kann also nicht bedeuten, dass er nichts rekrutiert, also warum die 0?

**Comment by Sh0wdown** [comment]
> Frage an @Sh0wdown: ist 0 schneller als 1? Emir beispielsweise hat auf **Default** und **Strong** einen Wert von 0. Kann also nicht bedeuten, dass er nichts rekrutiert, also warum die 0?

Gibt keinen Unterschied zwischen 0 und 1. Resultiert mit beiden Werten in dem allerhöchsten Rekrutiertempo (einmal pro AI routine).

**Comment by DarkThief-Darek** [comment]
Alls klar, danke für die schnelle Antwort. Komisch was sich Firefly dabei gedacht hat.

Ich möchte noch ausprobieren, ob die Werte 2, 3, 5, 6 und 7 wenigstens einen spürbaren Unterschied bringen zu den derzeitig benutzten Werten. 4 ist oftmals zu langsam für viele KIs, 1 aber zu schnell. 

Oder weißt du zufällig, ob diese Werte nichts ändern? @Sh0wdown

**Comment by Cenraves** [comment]
> > Vielen lieben Dank dafür! Ich wusste gar nicht, das es so etwas bereits im Spiel gibt. Ich habe bereits etwas rumprobiert, allerdings erreicht man diesen Stärkestatus leider viel zu einfach. Kann man Diesen denn irgendwie neu definieren?
> 
> Ja, allerdings nicht über AIC edits. Da müsste ich also eine neue Option für hinzufügen.

Das wäre ausgezeichnet! So könnte man den Wert der vorher unter "RecruitIntervalStrong" mit einer 1 oder 0 ersetzen und unter "RecruitInterval" diesen Wert stattdessen eintragen. Mit einer neuen Definition, mit der eine KI zum Beispiel 10000 Gold benötigt, würde sie niemals unter die 10000 Gold Grenze fallen (ohne Druck von außen) und hätte damit ein Polster für schwere Schäden. Zusätzlich wäre das sogar noch ein Nachteil für die KI, da sie von allen rausgefocust wird, die nach dem meisten Gold angreifen.

**Comment by ByBurton** [comment]
@Sh0wdown 
Das wäre durchaus cool wenn du irgendwie als Option hinzufügen würdest, dass die Grenzen zwischen diesen drei Zuständen offensichtlicher und weiter voneinander entfernt sind.
Der schwache Zustand wird praktisch nie erreicht, und der starke schon vieeel zu früh.
